src/rmind/components/episode.py

```diff
@@ -143,13 +142,14 @@ class Episode(TensorClass["frozen"]):
     input: TensorDict  # pyright: ignore[reportUninitializedInstanceVariable]
     input_tokens: TensorDict  # pyright: ignore[reportUninitializedInstanceVariable]
     input_embeddings: TensorDict  # pyright: ignore[reportUninitializedInstanceVariable]
+    projected_embeddings: TensorDict  # pyright: ignore[reportUninitializedInstanceVariable]
     position_embeddings: TensorDict  # pyright: ignore[reportUninitializedInstanceVariable]
     index: Index  # pyright: ignore[reportUninitializedInstanceVariable]
     timestep: Timestep  # pyright: ignore[reportUninitializedInstanceVariable]
 
     @property
     def embeddings(self) -> TensorDict:
-        return self.input_embeddings + self.position_embeddings
+        return self.projected_embeddings + self.position_embeddings
```

Comment:
it seems we still mask `input_embeddings` in `RandomMaskedHindsightControlObjective` but the masking has no effect since we don't re-project post masking and use the pre-masking `projected_embeddings` in `Episode.embeddings_packed`?